### PR TITLE
Make base installation more lightweight. 

### PR DESCRIPTION
**Pull request by aadyotb**
This PR allows for a more lightweight base installation of Merlion by making dependencies for Prophet & deep learning models be optional. It also adds helpful error messages when trying to use a model whose optional dependencies are not installed, and it updates the version to v.1.1.1.

### CODE REVIEW

File: merlion/models/forecast/lstm.py [hunk 1]
```diff
@@ -1,12 +1,24 @@
...
 A forecaster based on a LSTM neural net.
 """
+try:
+    import torch
+    import torch.nn as nn
+    import torch.nn.functional as F
+    from torch.utils.data import DataLoader, Dataset
+except ImportError as e:
+    err = (
+        "Try installing Merlion with optional dependencies using `pip install salesforce-merlion[torch]` or "
```
Comment (paulkass) [inline merlion/models/forecast/lstm.py:17]: Should this recommend `salesforce-merlino[deep-learning]` instead?

File: merlion/models/ensemble/MoE_networks.py [hunk 2]
```diff
@@ -1,12 +1,19 @@
...
-import torch
-import torch.nn.functional as F
-from torch import nn
+try:
+    import torch
+    import torch.nn.functional as F
+    from torch import nn
+except ImportError as e:
+    err = (
+        "Try installing Merlion with optional dependencies using `pip install salesforce-merlion[torch]` or "
```
Comment (paulkass) [inline merlion/models/ensemble/MoE_networks.py:13]: Similarly

File: merlion/models/ensemble/MoE_forecast.py [hunk 3]
```diff
@@ -1,18 +1,27 @@
...
 """Mixture of Expert forecasters."""
 __author__ = "Devansh Arpit"
+
+try:
+    import torch
+    from torch import nn
+    from torch.utils.data import Dataset
+except ImportError as e:
+    err = (
+        "Try installing Merlion with optional dependencies using `pip install salesforce-merlion[torch]` or "
```
Comment (paulkass) [inline merlion/models/ensemble/MoE_forecast.py:16]: similarly

File: merlion/models/anomaly/vae.py [hunk 4]
```diff
@@ -1,17 +1,25 @@
...
 """
-import torch
-import torch.nn as nn
+try:
+    import torch
+    import torch.nn as nn
+    from torch.utils.data import DataLoader
+except ImportError as e:
+    err = (
+        "Try installing Merlion with optional dependencies using `pip install salesforce-merlion[torch]` or "
```
Comment (paulkass) [inline merlion/models/anomaly/vae.py:16]: Might be worth exporting this somehow :D

File: merlion/models/anomaly/utils.py [hunk 5]
```diff
@@ -1,10 +1,18 @@
...
 # All rights reserved.
 # SPDX-License-Identifier: BSD-3-Clause
 # For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
 #
-import torch
+try:
+    import torch
+except ImportError as e:
+    err = (
+        "Try installing Merlion with optional dependencies using `pip install salesforce-merlion[torch]` or "
```
Comment (paulkass) [inline merlion/models/anomaly/utils.py:11]: similarly

File: merlion/models/anomaly/lstm_ed.py [hunk 6]
```diff
@@ -1,17 +1,25 @@
...
 """
 The LSTM-encoder-decoder-based anomaly detector for multivariate time series
 """
+try:
+    import torch
+    import torch.nn as nn
+    from torch.utils.data import DataLoader
+except ImportError as e:
+    err = (
+        "Try installing Merlion with optional dependencies using `pip install salesforce-merlion[torch]` or "
```
Comment (paulkass) [inline merlion/models/anomaly/lstm_ed.py:16]: similarly

File: merlion/models/anomaly/deep_point_anomaly_detector.py [hunk 7]
```diff
@@ -11,9 +11,17 @@
...
-import torch.nn as nn
-import torch.utils.data as data
+
+try:
+    import torch
+    import torch.nn as nn
+    import torch.utils.data as data
+except ImportError as e:
+    err = (
+        "Try installing Merlion with optional dependencies using `pip install salesforce-merlion[torch]` or "
```
Comment (paulkass) [inline merlion/models/anomaly/deep_point_anomaly_detector.py:21]: similarly

File: merlion/models/anomaly/dagmm.py [hunk 8]
```diff
@@ -1,17 +1,26 @@
...
 Deep autoencoding Gaussian mixture model for anomaly detection (DAGMM)
 """
+try:
+    import torch
+    import torch.nn as nn
+    import torch.nn.functional as F
+    from torch.utils.data import DataLoader
+except ImportError as e:
+    err = (
+        "Try installing Merlion with optional dependencies using `pip install salesforce-merlion[torch]` or "
```
Comment (paulkass) [inline merlion/models/anomaly/dagmm.py:17]: similarly

File: merlion/models/anomaly/autoencoder.py [hunk 9]
```diff
@@ -1,17 +1,25 @@
...
 """
-import torch
-import torch.nn as nn
+try:
+    import torch
+    import torch.nn as nn
+    from torch.utils.data import DataLoader
+except ImportError as e:
+    err = (
+        "Try installing Merlion with optional dependencies using `pip install salesforce-merlion[torch]` or "
```
Comment (paulkass) [inline merlion/models/anomaly/autoencoder.py:16]: similarly

**Review by paulkass — CHANGES_REQUESTED** [review]

File: merlion/models/forecast/lstm.py (hunk 1 above)
Comment (aadyotb) [reply merlion/models/forecast/lstm.py]: Thanks for the catch. Ditto @ below.

**Review by paulkass — APPROVED** [review]

**Pull request merged**